asm/c128/stubs_test.go

```diff
@@ -0,0 +1,231 @@
+		x:   []complex128{0, 0},
+		y:   []complex128{1, 1, 1},
+		ex:  []complex128{1, 1}},
+	{incX: 2, incY: 2, incDst: 3, ix: 0, iy: 0, idst: 0,
+		a:   1 + 2i,
+		dst: []complex128{1i, 1i, 1i},
+		x:   []complex128{1i, 1i, 1i},
+		y:   []complex128{1, 2, 1},
+		ex:  []complex128{-1 + 1i, 1i, -1 + 1i}},
+	{incX: 2, incY: 2, incDst: 3, ix: 0, iy: 0, idst: 0,
+		a:   -1i,
+		dst: []complex128{1i, 1i, 1i},
+		x:   []complex128{1i, 1i, 1i},
+		y:   []complex128{1, 2, 1},
+		ex:  []complex128{2, 3, 2}},
+	{incX: 2, incY: 2, incDst: 3, ix: 0, iy: 0, idst: 0,
+		a:   -1i,
+		dst: []complex128{1i, 1i, 1i},
+		x:   []complex128{1i, 1i, 1i, 1i, 1i}[1:4],
+		y:   []complex128{1, 1, 2, 1, 1}[1:4],
+		ex:  []complex128{2, 3, 2}},
+	{incX: 2, incY: 4, incDst: 3, ix: 0, iy: 0, idst: 0,
+		a:   -2,
+		dst: []complex128{1i, 1i, 1i, 1i, 1i},
+		x:   []complex128{2 + 1i, 2 + 1i, 2 + 1i, 2 + 1i, 2 + 1i},
+		y:   []complex128{1, 1, 2, 1, 1},
+		ex:  []complex128{-3 - 2i, -3 - 2i, -2 - 2i, -3 - 2i, -3 - 2i}},
+	// Run big test twice, once aligned once unaligned.
+	{incX: 2, incY: 2, incDst: 3, ix: 0, iy: 0, idst: 0,
+		a:   1 - 1i,
+		dst: make([]complex128, 10),
+		x:   []complex128{1i, 1i, 1i, 1i, 1i, 1i, 1i, 1i, 1i, 1i},
+		y:   []complex128{1, 1, 2, 1, 1, 1, 1, 2, 1, 1},
+		ex:  []complex128{2 + 1i, 2 + 1i, 3 + 1i, 2 + 1i, 2 + 1i, 2 + 1i, 2 + 1i, 3 + 1i, 2 + 1i, 2 + 1i}},
+	{incX: 2, incY: 2, incDst: 3, ix: 0, iy: 0, idst: 0,
+		a:   1 - 1i,
+		dst: make([]complex128, 10),
+		x:   []complex128{1i, 1i, 1i, 1i, 1i, 1i, 1i, 1i, 1i, 1i},
+		y:   []complex128{1, 1, 2, 1, 1, 1, 1, 2, 1, 1},
+		ex:  []complex128{2 + 1i, 2 + 1i, 3 + 1i, 2 + 1i, 2 + 1i, 2 + 1i, 2 + 1i, 3 + 1i, 2 + 1i, 2 + 1i}},
+	{incX: -2, incY: -2, incDst: -3, ix: 18, iy: 18, idst: 27,
+		a:   1 - 1i,
+		dst: make([]complex128, 10),
+		x:   []complex128{1i, 1i, 1i, 1i, 1i, 1i, 1i, 1i, 1i, 1i},
+		y:   []complex128{1, 1, 2, 1, 1, 1, 1, 2, 1, 1},
+		ex:  []complex128{2 + 1i, 2 + 1i, 3 + 1i, 2 + 1i, 2 + 1i, 2 + 1i, 2 + 1i, 3 + 1i, 2 + 1i, 2 + 1i}},
+	{incX: -2, incY: 2, incDst: -3, ix: 18, iy: 0, idst: 27,
+		a:   1 - 1i,
+		dst: make([]complex128, 10),
+		x:   []complex128{1i, 1i, 1i, 1i, 1i, 1i, 1i, 1i, 1i, 1i},
+		y:   []complex128{1, 1, 2, 1, 1, 1, 1, 2, 1, 1},
+		ex:  []complex128{2 + 1i, 2 + 1i, 3 + 1i, 2 + 1i, 2 + 1i, 2 + 1i, 2 + 1i, 3 + 1i, 2 + 1i, 2 + 1i}},
+}
+
+func guardVector(v []complex128, g complex128, g_ln int) (guarded []complex128) {
+	guarded = make([]complex128, len(v)+g_ln*2)
+	copy(guarded[g_ln:], v)
+	for i := 0; i < g_ln; i++ {
+		guarded[i] = g
+		guarded[len(guarded)-1-i] = g
+	}
+	return guarded
+}
+
+func isValidGuard(v []complex128, g complex128, g_ln int) bool {
+	for i := 0; i < g_ln; i++ {
+		if v[i] != g || v[len(v)-1-i] != g {
+			return false
+		}
+	}
+	return true
+}
+
+func TestAxpyUnitary(t *testing.T) {
+	var x_gd, y_gd complex128 = 1, 1
+	for j, v := range tests {
+		xg_ln, yg_ln := 4+j%2, 4+j%3
+		v.x, v.y = guardVector(v.x, x_gd, xg_ln), guardVector(v.y, y_gd, yg_ln)
```

Comment:
I'm not sure what you're suggesting here.  Calling `guardVector` and `guardIncVector` can only happen within the test loop.  Otherwise, we need to maintain an entirely separate table of tests for the increment variants of these same tests.